isis/python_bindings/apps/findFeaturesSegment/findFeaturesSegment.py

```diff
@@ -332,28 +344,48 @@ def findFeaturesSegment(ui):
     output = output.get()
     log.debug(f"{output}")
     
-    image_sets = list(itertools.product(match_segments.values(), from_segments.values())) 
-    
+    # Remove redundant overlapping pairs
+    image_sets = list(itertools.product(match_segments.values(), from_segments.values()))
+    x = match_segments.values()
+    y = from_segments.values()
+    x,y = (x,y) if len(x) > len(y) else (y,x)
+    image_cartesian = list(itertools.product(x, y))
+    image_sets = []
+    for i in image_cartesian:
+        if i[0][0]["Segment"] >= i[1]["Segment"]:
+            image_sets.append(i) 
```

Comment:
How was this tested? There is some difference between the example I gave in the issue and how it would be implemented here. Just want to make sure it is working.

Reply:
So I added some print statements with the old `image_sets` and `image_sets` after this for loop for the two areas Lynn mentioned in [here](https://github.com/DOI-USGS/ISIS3/issues/5489#issuecomment-2080072192) (I can point you to the data areas too) and the results are attached as [image_set_sort_before.txt](https://github.com/user-attachments/files/16431744/image_set_sort_before.txt) and [image_set_sort_after.txt](https://github.com/user-attachments/files/16431749/image_set_sort_after.txt). It looks like both examples show the removal of a redundant pair, going from `(1,1), (1,2), (2,1), (2,2)` to `(1,1), (2,1), (2,2)`.